src/probnum/randvars/_config.py

```diff
@@ -0,0 +1,10 @@
+from probnum import config
+
+
+def _register_default_config():
+    _defaults = [
+        ("covariance_inversion_damping", 1e-12),
+    ]
+
+    for k, v in _defaults:
+        config.register(k, v)
```

Comment:
what is the rationale behind registering this parameter from `randvars` and not in top-level? (I think this parameter should be more general, as it would apply to e.g. the Kalman filtering equations, but my question is also about "why not register all the parameters in top-level).

Reply:
is it because the parameter only becomes available once `randvars` is imported?

Reply:
thanks for the explanation! Is there anything that speaks against registering all parameters that are available "globally" in the toplevel `_config`? I thought that the potential module-wide configs refer to `with diffeq.config(do_something=True)`, i.e. with a separate config in diffeq. Maybe we misunderstood each other.

Reply:
I think we should try to only have one `config` object in which all subpackages inject their settings, i.e. for the `diffeq` config options you would add the settings to the global object, but the registration calls should still be made in `diffeq/_config.py`

Reply:
Why wouldnt I add the settings in the global config then? What is the advantage of doing it in diffeq (if we stay with this example)? (Not sure whether "global config" is the proper term, but you know what I mean.)

Reply:
One reason might be the tendency to produce merge conflicts when working in one large config file.
Another one is the possibility to not register a subpackage's options, should we decide in the future that we do not automatically import all subpackages in `probnum/__init__.py` (similar to scipy).

Moreover, this way we increase code locality while maintaining a single interface towards the end-user of the library

Reply:
But I'm not set on the idea of just having one config object.
I just fear that we end up producing a mess (from the perspective of an end-user).

Reply:
Hmmm okay I see what you mean. Thanks for explaining the thought process :) 

> One reason might be the tendency to produce merge conflicts when working in one large config file.

I see this as a potential problem, but I dont think the problem would be severe. I don't expect the config file to be touched often (though this is speculation, ofc). 

> Another one is the possibility to not register a subpackage's options, should we decide in the future that we do not automatically import all subpackages in probnum/__init__.py (similar to scipy).

Okay, but if all the configs are gathered in one file, it is easy to move them to the subpackages if this decision is made at some point, right?

> Moreover, this way we increase code locality while maintaining a single interface towards the end-user of the library

I agree that the latter is useful. But I personally think that not setting the parameters that are available globally right in the global scope will also make the implementation more tedious to work with; at least in the beginning (as in "where has this parameter been set? What does it mean?" or even "do we have this yet"). 
From a different angle: how would we try to enforce that some parameter that has been registered in `randvars` is not used somewhere else (we dont have to, but it might help avoiding a mess)?

In principle, these downsides could of  course be solved with more extensive docs, but maybe this is not necessary.

All in all, I see the thought process behind this -- thanks for explaining. I still think that registering all options in the global config (at least for now) is the simpler and perhaps more appropriate solution for the moment. Of course, this is subjective.